toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -152,6 +168,7 @@ func buildBuildNode(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, graphMu
 	}
 
 	usedCache = false
+	hadMissingFiles = len(missingFiles) > 0
```

Comment:
Small ask: can we move setting `hadMissingFiles` before the `if` above it, so we can replace the code duplication in `if len(missingFiles) > 0` with `if hasMissingFiles`?